============================= test session starts ==============================
platform darwin -- Python 3.12.2, pytest-8.1.0, pluggy-1.4.0
rootdir: /Users/sidchaini/Research/DistClassiPy
configfile: pyproject.toml
plugins: hypothesis-6.100.7, anyio-4.3.0, cov-4.1.0
collected 69 items

tests/test_classifier.py .................                               [ 24%]
tests/test_distances.py ................................................ [ 94%]
....                                                                     [100%]

=============================== warnings summary ===============================
tests/test_classifier.py: 10 warnings
  /Users/sidchaini/miniconda3/lib/python3.12/site-packages/numpy/core/_methods.py:206: RuntimeWarning: Degrees of freedom <= 0 for slice
    ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,

tests/test_classifier.py: 10 warnings
  /Users/sidchaini/miniconda3/lib/python3.12/site-packages/numpy/core/_methods.py:195: RuntimeWarning: invalid value encountered in divide
    ret = um.true_divide(

tests/test_distances.py: 300 warnings
  /Users/sidchaini/Research/DistClassiPy/distclassipy/distances.py:263: RuntimeWarning: The correlation metric defaults to 0 when used on 1D vectors
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform darwin, python 3.12.2-final-0 ----------
Coverage XML written to file coverage.xml

====================== 69 passed, 320 warnings in 34.66s =======================
